src/java/com/rapleaf/jack/ModelField.java

```diff
@@ -0,0 +1,61 @@
+package com.rapleaf.jack;
+
+import com.rapleaf.jack.queries.Utility;
+
+public class ModelField {
+  private static String DEFAULT_ID_FIELD = "id";
+
+  private final Class<? extends ModelWithId> model;
+  private final Enum field;
+  private final Class type;
+
+  private ModelField(Class<? extends ModelWithId> model, Enum field, Class type) {
+    this.model = model;
+    this.field = field;
+    this.type = type;
+  }
+
+  public static ModelField key(Class<? extends ModelWithId> model) {
+    return field(model, null, Long.class);
+  }
+
+  public static ModelField field(Class<? extends ModelWithId> model, Enum field, Class fieldType) {
+    return new ModelField(model, field, fieldType);
+  }
+
+  public Class<? extends ModelWithId> getModel() {
+    return model;
+  }
+
+  public Enum getField() {
+    return field;
+  }
+
+  public Class getType() {
+    return type;
+  }
+
+  public String getSqlKeyword() {
+    String fieldKeyword = field != null ? field.toString() : DEFAULT_ID_FIELD;
```

Comment:
Is "null" for id just necessary because of how we use ModelWithId and it isn't an Attribute? Do you feel this is something we should change at some point? Ideally I'd like to have ModelWithId just be a specific type of Model and many usages, like this one, wouldn't need to rely on the WithId part.